addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -301,41 +296,25 @@ class ProductTemplate(models.Model):
         string='Procurement', default='rfq')
 
 
-class ProcurementOrder(models.Model):
-    _inherit = 'procurement.order'
-
-    requisition_id = fields.Many2one('purchase.requisition', string='Latest Requisition')
+class ProcurementRule(models.Model):
+    _inherit = 'procurement.rule'
 
     @api.multi
-    def make_po(self):
-        Requisition = self.env['purchase.requisition']
-        procurements = self.env['procurement.order']
-        Warehouse = self.env['stock.warehouse']
-        res = []
-        for procurement in self:
-            if procurement.product_id.purchase_requisition == 'tenders':
-                warehouse_id = Warehouse.search([('company_id', '=', procurement.company_id.id)], limit=1).id
-                requisition_id = Requisition.create({
-                    'origin': procurement.origin,
-                    'date_end': procurement.date_planned,
-                    'warehouse_id': warehouse_id,
-                    'company_id': procurement.company_id.id,
-                    'procurement_id': procurement.id,
-                    'picking_type_id': procurement.rule_id.picking_type_id.id,
-                    'line_ids': [(0, 0, {
-                        'product_id': procurement.product_id.id,
-                        'product_uom_id': procurement.product_uom.id,
-                        'product_qty': procurement.product_qty
-                    })],
-                })
-                procurement.message_post(body=_("Purchase Requisition created"))
-                requisition_id.message_post_with_view('mail.message_origin_link',
-                    values={'self': requisition_id, 'origin': procurement},
-                    subtype_id=self.env['ir.model.data'].xmlid_to_res_id('mail.mt_note'))
-                procurement.requisition_id = requisition_id
-                procurements += procurement
-                res += [procurement.id]
-        set_others = self - procurements
-        if set_others:
-            res += super(ProcurementOrder, set_others).make_po()
-        return res
+    def _run_buy(self, product_id, product_qty, product_uom, location_id, name, origin, values):
+        if product_id.purchase_requisition != 'tenders':
+            return super(ProcurementRule, self)._run_buy(product_id, product_qty, product_uom, location_id, name, origin, values)
+
+        self.env['purchase.requisition'].create({
```

Comment:
Why do you want a _prepare_tender_values? Since _run_buy will probably be called by a move in order to create the PO you can override the _prepare_procurement_values(). Or a better solution is to directly override the _prepare_purchase_order_line() method in order to add the key you want.

For the second point It's a improvement so I will keep as a feedback for the moment and see what is it possible to do with the product owner team about it. (No for the moment since we have a lot to do with the v11)

Reply:
Well, there can be possible extensions to the tender model that can't be attended in `_prepare_procurement_values`. For example, new fields in it, that needs to be filled according values dictionary or maybe other criteria.

For the second point, I understand that is something not feasible for v11, so just take into account as a suggestion.

Reply:
Ok sorry got it.